src/compiler/compile/nodes/Animation.ts

```diff
@@ -31,7 +34,7 @@ export default class Animation extends Node {
 			});
 		}
 
-		block.has_animation = true;
+		(block as any).has_animation = true;
```

Comment:
Is `block` `EachBlock` here? If I add a return or throw after `component.error`, typescript's type guard will infer it as `EachBlock` here. Wondering if we can change or wrap `component.error` to an [assertion function](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions) once we bump typescript version so that we can rely on type guard in similar situations.

Reply:
Oh, that's true. I didn't see that it's supposed to be an `EachBlock`. I'll change the cast to that. Thanks! Guess we can't do much more in the current TS version.